airflow/models/taskinstance.py

```diff
@@ -872,6 +1163,8 @@ def _is_eligible_to_retry(*, task_instance: TaskInstance | TaskInstancePydantic)
     return task_instance.task.retries and task_instance.try_number <= task_instance.max_tries
 
 
+@provide_session
+@internal_api_call
 def _handle_failure(
```

Comment:
as above - make naming distinct.
```suggestion
def _handle_ti_failure(
```